lib/compress/zstd_compress.c

```diff
@@ -1599,20 +1599,41 @@ static size_t ZSTD_compress_frameChunk (ZSTD_CCtx* cctx,
             return ERROR(dstSize_tooSmall);   /* not enough space to store compressed block */
         if (remaining < blockSize) blockSize = remaining;
 
-        /* preemptive overflow correction */
+        /* preemptive overflow correction:
+         * 1. correction is large enough:
+         *    lowLimit > (3<<29) ==> current > 3<<29 + 1<<windowLog - blockSize
```

Comment:
It shows that overflow correction is driven by the value of `lowLimit`.
This is true because currently `lowLimit` is constantly updated to ensure that no offset beyond `windowSize` can be valid, effectively acting as a "free" offset checker.

In the future, if we implement a "real" offset checker, notably to better serve small window sizes, it will no longer be necessary to update `lowLimit` to do the "offset validation". 
But it will still be necessary to update it to make sure we trigger overflow correction.

Nothing to be worried about at this stage, just something to remember in the future.